packages/calypso-e2e/src/lib/components/editor-settings-sidebar-component.ts

```diff
@@ -31,9 +31,8 @@ const selectors = {
 
 	// Schedule
 	scheduleButton: `button.edit-post-post-schedule__toggle`,
-	scheduleInput: ( attribute: string ) => `role=spinbutton[name="${ attribute }"i]`,
+	scheduleInput: ( name: string ) => `.edit-post-post-schedule__dialog label:has-text("${ name }")`,
```

Comment:
**comment**: this may be slightly out of scope for this change and it is up to you whether you'd want to make the change.

@WunderBart has been replacing many selector templates like this with [role-based locators](https://playwright.dev/docs/locators) and I am a big fan of the change.

For instance, instead of having this CSS selector we can remove `scheduleInput` entirely, replacing its usage as follows:

```
await page.getByLabel('Day')

or

await page.getByLabel('Month')
```

right in the function call itself. It's one way to remove the need to constantly maintain selectors.

Reply:
That sounds good! The only downside is that they are not scoped to the post schedule dialog, so there's a small risk that it could select the wrong element if the page ever has two inputs with the same Label (which I think is unlikely in this context, in practice).

Reply:
I'll take note of that and do it in a small follow-up soon.